### ISSUE TITLE
change variables in queries for references

### ISSUE THREAD

**Issue by yehosef** [elastic/timelion#43]
Sometimes I have similar changes with a small change in the query.  

.es(index="users" q="+signup_date:[2015-12-01 TO 2015-12-31] +action:save")

And I have another chart that needs the same thing but with "+action:edit",

It would be great if I could do something like:

$action="save", .es(index="users" q="+signup_date:[2015-12-01 TO 2015-12-31] +action:$action")

and then in a different chart use (@1, $action="edit")

Additionally, it would be nice to be able to override basic components of the reference like:
(@1 index="users-new") 

or similar.

**Comment by rashidkpc** [comment]
Yeah, there's no string concatenation. I purposefully avoided documenting variables because I wasn't sure I wanted to keep that construct.

**Comment by rashidkpc** [comment]
If I do string interpolation it will probably be ES6 style, eg backticks + ${} so in backticks:

`this is my ${string}`
